### ISSUE TITLE
Standardize on torch.inference_mode() for all inference paths

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#5663]
# Context7 Best Practices Audit Finding

## Current Implementation

Inconsistent usage of inference contexts:

```python
# ai/warmup_utils.py - Mixed usage
with torch.inference_mode():  # line 287, 300
with torch.no_grad():  # line 177, 411

# ai/batch_utils.py:458
with torch.no_grad():
```

## Best Practice (Context7)

`torch.inference_mode()` provides more aggressive optimization than `torch.no_grad()`:

* Disables view tracking
* Disables version counting
* Better memory efficiency
* Slightly faster execution

## Gap Analysis

* Inconsistent usage across codebase
* Missing potential performance gains
* Code style inconsistency

## Recommended Changes

Standardize on `torch.inference_mode()` for all inference-only code:

```python
# Before
with torch.no_grad():
    output = model(input)

# After
with torch.inference_mode():
    output = model(input)
```

Reserve `torch.no_grad()` only when:

* Backward compatibility with PyTorch < 1.9 needed
* Code may need gradients later (fine-tuning paths)

## Files to Modify

* `ai/warmup_utils.py` - Replace torch.no_grad() with torch.inference_mode()
* `ai/batch_utils.py` - Replace torch.no_grad() with torch.inference_mode()
* Search for other occurrences across codebase

## Acceptance Criteria

- [ ] All inference-only paths use torch.inference_mode()
- [ ] torch.no_grad() reserved for specific documented cases
- [ ] Linting rule to enforce consistency (optional)
